tests/unit/helpers.py

```diff
@@ -43,6 +43,20 @@ def make_client(project="PROJECT", **kw):
     return google.cloud.bigquery.client.Client(project, credentials, **kw)
 
 
+def make_creds(creds_universe: None):
+    from google.auth import credentials
+
+    class TestingCreds(credentials.Credentials):
+        def refresh(self, request):  # pragma: NO COVER
```

Comment:
For my education, what is the purpose of the `refresh()` method here?

Reply:
This subclasses credentials.  Refresh is part of the mechanism to provide bearer credentials, but is marked abstract in the base class so we need to provide an impl even though we're not using it here.